### PR TITLE
Apr 1337 add subcontractor declatation gateway page

### PR DESCRIPTION
**Pull request by MichaelYoung1981**

### CODE REVIEW

File: src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/LegalNameTests.cs [hunk 1]
```diff
@@ -13,7 +14,7 @@
 namespace SFA.DAS.AdminService.Web.Tests.Services.Gateway.OrganisationChecks.Orchestrator
 {
     [TestFixture]
-    public class LegalNameTests
+    public class SubcontractorDeclarationTests
```
Comment (MarkFCain) [inline src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/LegalNameTests.cs]: This name change looks wrong

Comment (MichaelYoung1981) [reply src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/OrganisationChecks/Orchestrator/LegalNameTests.cs]: I'm blaming ReSharper! 😁 Reverted that change.

File: src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayExperienceAndAccreditationOrchestrator.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+            return model;
+        }
+
+        public async Task<FileStreamResult> GetSubcontractorDeclarationContractFile(GetSubcontractorDeclarationContractFileRequest request)
+        {
+            return await _experienceAndAccreditationApiClient.GetSubcontractorDeclarationContractFile(request.ApplicationId);
+        }
+    }
+}
```
Comment (MarkFCain) [inline src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayExperienceAndAccreditationOrchestrator.cs]: Cant see any tests for this.  There's a lot of tests kicking around though.... Please add test if needed

Comment (MichaelYoung1981) [reply src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayExperienceAndAccreditationOrchestrator.cs]: SFA.DAS.AdminService.Web.Tests.Services.Gateway.ExperienceAndAccreditation.SubcontractorDeclarationTests
Isn't clear that class relates to the orchestrator tho so I'll rename it.

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/OrganisationChecks/IcoNumberTests.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        [SetUp]
+        public void Setup()
+        {
+            _applyApiClient = new Mock<IRoatpApplicationApiClient>();
+            _contextAccessor = new Mock<IHttpContextAccessor>();
+            _gatewayValidator = new Mock<IRoatpGatewayPageViewModelValidator>();
+            _logger = new Mock<ILogger<RoatpGatewayOrganisationChecksController>>();
+            _orchestrator = new Mock<IGatewayOrganisationChecksOrchestrator>();
+
+            var user = new ClaimsPrincipal(new ClaimsIdentity(new Claim[]
```
Comment (David-B-Read) [inline src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/OrganisationChecks/IcoNumberTests.cs]: We do this setup in a lot of unit test classes now, potential for this to be moved into a base class?

Comment (MichaelYoung1981) [reply src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/OrganisationChecks/IcoNumberTests.cs]: Definitely.  I'll need to do it on my next page, happy if I do it then?  I'll go back and change any existing tests as well.

Comment (David-B-Read) [reply src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/OrganisationChecks/IcoNumberTests.cs]: Yep, works for me.

**Pull request merged**